### PR TITLE
Rewrites simple modules to TypeScript

### PR DESCRIPTION
**Pull request by artem-zakharchenko**
#### :rocket: Why this change?

- To slowly begin to rewrite Dredd internal modules to TypeScript
- Rewrites and simplifies `sortTransactions` module

#### :memo: Related issues and Pull Requests

- Follows #1524 

#### :white_check_mark: What didn't I forget?

- [x] To write docs
- [ ] To write tests
- [x] To put [Conventional Changelog](https://dredd.org/en/latest/internals.html#sem-rel) prefixes in front of all my commits and run `npm run lint`
- [x] Add unit tests for `sortTransactions` module

### CODE REVIEW

File: packages/dredd/lib/general.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    bodyEncoding: BodyEncoding;
+  };
+  skip: boolean;
+  fail: boolean;
+
+  test: TransactionTest;
+}
+
+export interface TransactionRequest {
+  method: RESTMethod;
```
Comment (honzajavorek) [inline packages/dredd/lib/general.ts]:
```suggestion
  method: HTTPMethod;
```

File: packages/dredd/lib/general.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
+export enum RESTMethod {
```
Comment (honzajavorek) [inline packages/dredd/lib/general.ts]:
```suggestion
export enum HTTPMethod {
```

File: packages/dredd/test/unit/sortTransactions-test.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
+import R from 'ramda';
+import { expect } from 'chai';
+import sortTransactions from '../../lib/sortTransactions';
+import { Transaction, RESTMethod } from '../../lib/general';
+
```
Comment (honzajavorek) [inline packages/dredd/test/unit/sortTransactions-test.ts]: I don't see a similar test written in JS being removed. Does it mean you want to keep it to be sure nothing got broken, or that it doesn't exist and this part isn't tested?

**Comment by honzajavorek** [comment]
If you add two methods, you can close this one https://github.com/apiaryio/dredd/issues/267

File: packages/dredd/test/unit/sortTransactions-test.ts (hunk 3 above)
Comment (artem-zakharchenko) [reply packages/dredd/test/unit/sortTransactions-test.ts]: I couldn't find any existing test that would cover `sortTransactions`. To my knowledge this adds the missing unit test, so there is no JavaScript predecessor to remove. Please, point me if you know where this behavior may be tested already. Thanks.

File: packages/dredd/lib/general.ts (hunk 2 above)
Comment (artem-zakharchenko) [reply packages/dredd/lib/general.ts]: Resolved in 875fc45.

**Comment by artem-zakharchenko** [comment]
@honzajavorek thanks for the review.

Are you speaking about adding `LINK` and `UNLINK` to `HTTPMethod`? Are those even valid methods per spec?

**Comment by honzajavorek** [comment]
They are used with proxies.

**Comment by artem-zakharchenko** [comment]
@honzajavorek I think we can cover LINK/UNLINK separately. Thanks!

**Pull request merged**